### PR TITLE
Fix/timeseries xticks

### LINKED ISSUE #19: Time Series charts shows all X-ticks
Charts on mixed-content don't look properly formatted because it's rendering every x-tick.

### PR DESCRIPTION
**Pull request by JonathanChavezTamales**
### Summary

If merged, this PR will close #19. It adds a new query parameter to the time series endpoints. This PR is dependent on #16.

This new parameter is *datapoints*, it will select only a subset of the query datapoints proportionally spaced so that when plotting, they will not be one on top of the other.

Usage:
`http://localhost:4000/api/mixed-content/https-percentage-requests?datapoints=3`

It will respond with just 3 datapoints for each client.
```
{"description":"Time Series of the percentage of all resources that load with HTTPS","result":
[{"date":"2016_01_01","timestamp":1451606400000,"client":"mobile","percent":21.81},
{"date":"2017_04_01","timestamp":1491004800000,"client":"mobile","percent":43.48},
{"date":"2018_07_01","timestamp":1530403200000,"client":"mobile","percent":68.15},
{"date":"2016_01_01","timestamp":1451606400000,"client":"desktop","percent":24.04},
{"date":"2017_04_01","timestamp":1491004800000,"client":"desktop","percent":43.49},
{"date":"2018_07_01","timestamp":1530403200000,"client":"desktop","percent":74.89}],
"suggestedVisualizations":["Line chart"]}
```
### Tests
I also added two more tests for this new parameter
- [x] Check with 3 datapoints /https-percentage-pages
- [x] Check with more datapoints than they exist /https-percentage-pages

### CODE REVIEW

**Comment by ernestognw** [comment]
@googlebot I consent

**Comment by ernestognw** [comment]
Also merged master into this branch to avoid conflicts with Sofia's recent merge

File: server/api/mixed-content/index.js [hunk 1]
```diff
@@ -104,6 +117,18 @@ mixedApi.get('/https-percentage-requests', async (req, res) => {
   let rows = [];
   rows = await queryData(query);
 
+
+  // If there is a certain number of datapoints required for visualization
+  // it will select a subset with proportional skips.
+  const datapoints = req.query.datapoints;
+  if (datapoints) {
+    let mobileRows = rows.filter((row) => row.client === 'mobile');
```
Comment (ernestognw) [inline server/api/mixed-content/index.js:125]: nit: Would be better if we use .reduce here to filter both mobile and desktop in only one loop. I think datapoints is the kind of thing that can grow significantly in the future

File: server/api/mixed-content/index.js [hunk 2]
```diff
@@ -180,6 +205,30 @@ const queryData = async ({sql}) => {
...
 /**
+ * Selects a subset of elements of an array proportionally spaced
+ *
+ * If elements is 3 in this array, it selects: [1],2,3,[4],5,6,[7],8,9
+ *
+ * @param {Array} array Array to get subset from.
+ * @param {number} elements Total number of elements to retrieve.
+ * @return {Array} Subset of elements proportionally spaced.
+ */
+const select = (array, elements) => {
```
Comment (livvielin) [inline server/api/mixed-content/index.js:216]: Can we just do something like:
```
const newArray = [];
for (let i = 0; i < array.length && newArray.length < elements; i += Math.floor(array.length / elements)) {
    newArray.push(array[i]);
}
```

[the remark this answers is not in the document]
Comment (JonathanChavezTamales) [reply server/api/mixed-content/index.js:216]: Sure, that makes sense, converted into an issue (enhancement).

Comment (JonathanChavezTamales) [reply server/api/mixed-content/index.js:216]: Done, new commit is coming!

File: server/api/mixed-content/index.js [hunk 3]
```diff
@@ -180,6 +205,30 @@ const queryData = async ({sql}) => {
...
+  let skip = Math.floor(array.length/elements);
+  skip = skip == 0 ? 1 : skip;
+  let count =0;
+  const newArray = [];
+  for (let i=0; i<array.length; i+=skip) {
```
[the remark this answers is not in the document]
Comment (JonathanChavezTamales) [reply server/api/mixed-content/index.js]: done

File: server/api/mixed-content/index.js (hunk 1 above)
Comment (JonathanChavezTamales) [reply server/api/mixed-content/index.js:125]: Great, I think we can treat this as a P2 to keep working on what is left and later we try to optimize it.

**Pull request merged**